### PR TITLE
feat: add basic comment context formatting

### LINKED ISSUE #2691: Unformatted comment spans add extra newline
```rust
script;

// use statements
use std::*;
```
formats into:
```rust
script;


// use statements
use std::*;
```
Each additional format call adds more extra newlines
2nd format:
```rust
script;



// use statements
use std::*;
```

### LINKED ISSUE #2696: Add comment context formatting to sway-fmt-v2
Just like `rustfmt`, `swayfmt` should be able to cut down extra newline between comment blocks.

```rust
// This is a comment



// This is a comment
let number = 10;
```

should become something like

```rust
// This is a comment

// This is a comment
let number = 10;
```

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #2696.
closes #2691.

### CODE REVIEW

File: sway-fmt-v2/src/utils/map/comments.rs [hunk 1]
```diff
@@ -180,6 +180,22 @@ fn get_comments_between_spans(
     comments_with_context
 }
 
+fn format_context(context: &str, threshold: usize) -> String {
+    let mut remaning_newlines = threshold;
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/utils/map/comments.rs:184]:
```suggestion
    let mut remaining_newlines = threshold;
```

File: sway-fmt-v2/src/utils/map/comments.rs [hunk 2]
```diff
@@ -180,6 +180,22 @@ fn get_comments_between_spans(
...
+    let mut formatted_context = String::new();
+    for char in context.chars() {
+        if char == '\n' {
+            if remaning_newlines > 0 {
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/utils/map/comments.rs:188]:
```suggestion
            if remaining_newlines > 0 {
```

File: sway-fmt-v2/src/utils/map/comments.rs [hunk 3]
```diff
@@ -180,6 +180,22 @@ fn get_comments_between_spans(
...
+                formatted_context.push('\n');
+                remaning_newlines -= 1;
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/utils/map/comments.rs:190]:
```suggestion
                remaining_newlines -= 1;
```

**Review by mitchmindtree — APPROVED** [review]
Nice work on tracking this down :+1: I think these patches are definitely starting to get a little hacky, but we can do some cleanup in formatter once we've at least got all tests passing and have landed #2669.

**Review by bingcicle — APPROVED** [review]
utACK

**Pull request merged**